dev_docs/tutorials/adding_performance_metrics.mdx

```diff
@@ -0,0 +1,68 @@
+  eventName: KIBANA_LOADED_EVENT,
+  meta: {
+    kibana_version: this.coreContext.env.packageInfo.version,
+    protocol      : window.location.protocol,
+    ...fetchOptionalMemoryInfo(),
+  },
+  duration: timing[LOAD_FIRST_NAV],
+  key1    : LOAD_START,
+  value1  : timing[LOAD_START],
+  key2    : LOAD_BOOTSTRAP_START,
+  value2  : timing[LOAD_BOOTSTRAP_START],
+  key3    : LOAD_CORE_CREATED,
+  value3  : timing[LOAD_CORE_CREATED],
+  key4    : LOAD_SETUP_DONE,
+  value4  : timing[LOAD_SETUP_DONE],
+  key5    : LOAD_START_DONE,
+  value5  : timing[LOAD_START_DONE],
+});
+```
+
+Normally reporting an event requires a registration of event type however since we want
+performance metrics to be uniform across all kibana we already summarized it with the structure above.
+We welcome all the feedback based on your usecases.
+
+# Guidelines of What to report in Performance Metric Events
+
+
+In order to simplify the consumption of performance telemetry, we'd like to standardize the shape of events. This requires considering the following aspects:
+- `Performance` - amount and size of events being sent
+- `Discoverability` - how easy it is to find different events, dimensions and metrics
+- `Queriability` - we would like to have the ability to write complex queries on top of telemetry data; for example getting all dashboard load events that took over 10 seconds for dashboards that contain less than 5 panels.
+- `Flexibility` - Teams should be able to flexibly design their event structure without having to define custom mappings (except for edge cases). I.e. a team can decide that the first free field is always used for that page load.
+- `Ease of visualization` - it should be easy to visualize the data
+- `Index field explosion` - having too many field names in the index, damages the performance when visualizing the data.
+
+And event schema is composed of fields which is designed for specific purposes:
+
+- `Event name` We decided to report all metric events under a single event_type, so that we can avoid registering it multiple times. This requires us to add a special standard field for eventName.
+- `Standardized fields` (duration) these fields represent common values that might be reported for performance. This can include duration for time based reports, status, heapSize, memorySize, etc. Standardized fields will be mapped as is to the index, so they can be visualized.
+- `Free fields` (key1,value1,key2, value2, ..., key5, value5) A limited set of numbered free key-value pairs that can be used to report additional use case specific fields. Teams may use them to create their own internal sub-schema. For example, a team may choose to use key1 to always report the loading time of a certain component that is always present on the page. Free fields will be mapped, if present, to the index, so they can be visualized. If a free field becomes popular, it can be promoted to a standardized field, freeing up the free field for other use.
```

Comment:
nit for:
- consistent comma-delimited separation 😇 
- `integer-numbered` because we mapped them as `long`

```suggestion
- `Free fields` (key1, value1, key2, value2, ..., key5, value5) A limited set of integer-numbered free key-value pairs that can be used to report additional use-case specific fields. Teams may use them to create their own internal sub-schema. For example, a team may choose to use key1 to always report the loading time of a certain component that is always present on the page. Free fields will be mapped, if present, to the index, so they can be visualized. If a free field becomes popular, it can be promoted to a standardized field, freeing up the free field for other use.
```

---------------------

> If a free field becomes popular, it can be promoted to a standardized field, freeing up the free field for other use.

Any idea about how would the migration look like so existing visualizations are not broken with that change?

Reply:
I don't think we have clear strategy for this one but I'm assuming that we can suggest some type of visualisation versioning (in separate indices or same index but different visualization using time filter before/after the change) maybe which will let devs to have none breaking visualizations but it's quite hypothetical approach that I'm suggesting not sure if it has real world usecases on Kibana. Also the drawback of this approach we start to update the schema frequently then they will end up having bunch of visualisations with different versions. Do you have some bullet proof approach that you can suggest ? @afharo